library/src/main/java/com/wultra/android/sslpinning/CertStore.kt

```diff
@@ -265,7 +276,36 @@ class CertStore internal constructor(private val configuration: CertStoreConfigu
         }
     }
 
-    private fun processReceivedData(data: ByteArray, currentDate: Date): UpdateResult {
+    private fun processReceivedData(data: ByteArray, challenge: String?, responseHeaders: Map<String, String>, currentDate: Date): UpdateResult {
+
+        val publicKey = cryptoProvider.importECPublicKey(publicKey = configuration.publicKey)
+                ?: throw IllegalArgumentException("Illegal configuration public key")
+
+        // Validate signature in header
+        if (configuration.useChallenge) {
```

Comment:
If the library is configured for the challenge but the server doesn't return it, it will crash.

Reply:
Ok, it's safeguarded in `doUpdate` method, my bad..

Reply:
I made that exception more clear by adding an explanatory comment.